app/controllers/AlertController.java

```diff
@@ -61,6 +70,53 @@ public AlertController(final Configuration configuration, final AlertRepository
     }
 
     /**
+     * Adds or updates an alert in the alert repository.
+     * If a valid UUID is passed, it will update the corresponding alert if it exists or create an alert with this id.
+     * If no id is passed, it will generate an id.
+     *
+     * @param alertId The <code>String</code> representation of the id of the alert to be updated.
+     * @return Success if the alert was created or updated successfully, an error code otherwise.
+     */
+    public Result addOrUpdate(final String alertId) {
+        UUID id = UUID.randomUUID();
+        try {
+            if (alertId != null) {
+                id = UUID.fromString(alertId);
+            }
+        } catch (final IllegalArgumentException e) {
+            return badRequest("Invalid alert id");
+        }
+        final JsonNode jsonBody = request().body().asJson();
+        if (jsonBody == null) {
+            return badRequest("Request must have a non-empty body");
+        }
+        Alert alert;
+        try {
+            final DefaultAlert.Builder alertBuilder = OBJECT_MAPPER.readValue(jsonBody.toString(), DefaultAlert.Builder.class);
```

Comment:
You shouldn't be deserializing directly from the view model into the internal model.